core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStore.java

```diff
@@ -272,10 +272,11 @@ private void createSchemaTopic() throws StoreInitializationException {
   }
 
   static List<String> brokersToEndpoints(List<Broker> brokers) {
-    List<String> endpoints = new LinkedList<String>();
+    List<String> endpoints = new LinkedList<>();
     for (Broker broker : brokers) {
       for (EndPoint ep : JavaConversions.asJavaCollection(broker.endPoints())) {
-        endpoints.add(ep.connectionString());
+        String hostport = ep.host() == null ? ":" + ep.port() : Utils.formatAddress(ep.host(), ep.port());
+        endpoints.add(ep.securityProtocol() + "://" + hostport);
```

Comment:
Yeah, I'm thinking we probably did have this problem even before. We could just not validate it and document that you have to choose a single security protocol (which I guess is implied by `kafkastore.security.protocol` and is why the tests were still passing -- I think they all test clusters with only one protocol enabled).

re: parsing, it seems like the regex doesn't seem to match what is suggested in the comment. The initial `.*?` will match any prefix as long as the remainder of the string matches the `host:port` format. It's a reluctant match so you won't, e.g., incorrectly eat into the hostname if there isn't any prefix, but it allows arbitrary prefixes that terminate with an invalid char for the hostname section.

I'm actually inclined to at least leave the prefixes in. I think they end up being valuable debugging info if something is going wrong with your setup -- if you see mixed protocols or a mismatch with `security.protocol`, you've got a good idea where the source of the problem is (and it's still an improvement over the previous version which uses broker-specific info to this client).

I've pushed an update which now does filtering + warning based on the security.protocol setting, let me know if you think things now make sense.